lib/client/api.go

```diff
@@ -1823,8 +1847,12 @@ func (tc *TeleportClient) runShellOrCommandOnSingleNode(ctx context.Context, clt
 	return trace.Wrap(nodeClient.RunInteractiveShell(ctx, types.SessionPeerMode, nil, nil))
 }
 
-func (tc *TeleportClient) runShellOrCommandOnMultipleNodes(ctx context.Context, clt *ClusterClient, nodeAddrs []string, command []string) error {
+func (tc *TeleportClient) runShellOrCommandOnMultipleNodes(ctx context.Context, clt *ClusterClient, nodes []targetNode, command []string) error {
 	cluster := clt.ClusterName()
+	nodeAddrs := make([]string, 0, len(nodes))
```

Comment:
It looks like you only ever use `nodeAddrs[0]`. Do you need to build up a slice which could potentially contain many addrs if you only need the first one?

Reply:
We use `nodeAddrs` as a whole in the tracing span.
https://github.com/gravitational/teleport/blob/8a954c42abe28c6cde76d0bb65c2156ee4c4f3e3/lib/client/api.go#L1856-L1865